main.tf

```diff
@@ -25,6 +28,30 @@ resource "ibm_is_vpc" "vpc" {
 
   dns {
     enable_hub = var.enable_hub
+    # Creates a delegated resolver. Requires dns.enable_hub to be false.
+    resolver {
```

Comment:
I think using a dynamic block would make the code more readable here and avoid the repeats.

Also perhaps double check that having a resolver of type null is not going to throw an error.

Reply:
no it won't throw an error, have a look at the below findings.